frontend/components/user-settings/user-settings.tsx

```diff
@@ -0,0 +1,298 @@
+    const handleProfilePictureChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const file = e.target.files?.[0]
+        if (file) {
+        const reader = new FileReader()
+        reader.onloadend = () => {
+            setProfilePicture(reader.result as string)
+        }
+        reader.readAsDataURL(file)
+        }
+    }
+
+    // Function to delete the current profile picture, defaulting to a default placeholder image
+    const handleDeleteProfilePicture = () => {
+        setProfilePicture('/img/placeholder.svg?height=100&width=100')
+    }
+
+    // Function to save any updates to the user's information
+    const handleSaveChanges = async () => {
+        if (user) {
+        const token = localStorage.getItem('jwtToken')
+        if (!token) {
+            console.error('No authentication token found')
+            return
+        }
+
+        try {
+            const response = await fetch(`http://localhost:3001/users/${userId}`, {
+            method: 'PATCH',
+            headers: {
+                'Authorization': `Bearer ${token}`,
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(user),
+            })
+            if (!response.ok) {
+                throw new Error('Failed to save changes')
+            } else {
+                toast({
+                    title: "Success 💪",
+                    description: "User information is updated successfully!",
+                })
+            }
+
+            console.log('Changes saved successfully!')
+            mutate()
+        } catch (error) {
+            console.error('Error saving changes:', error)
+            toast({
+                title: "Failed ❗",
+                description: "An error has occured when saving changes.",
+            })
+        }
+        }
+    }
+
+
+    // Function to handle the deletion of the user account, which is called from the delete account modal
+    const handleDeleteAccount = async () => {
+        const token = localStorage.getItem('jwtToken')
+        if (!token) {
+            console.error('No authentication token found')
+            return
+        }
+
+        try {
+            const response = await fetch(`http://localhost:3001/users/${userId}`, {
+                method: 'DELETE',
+                headers: {
+                    'Authorization': `Bearer ${token}`,
+                    'Content-Type': 'application/json',
+                },
+            })
+            if (!response.ok) throw new Error('Failed to delete account')
+
+            console.log('Account deleted successfully!')
+            // Redirect to the main page after successful deletion
+            window.location.href = '/';
+        } catch (error) {
+            console.error('Error deleting account:', error)
+        }
+    }
+
+    if (isLoading) {
+        return <div>Loading user data...</div>
```

Comment:
We can consider using the [suspense](https://react.dev/reference/react/Suspense) component 

I think it might be nice to have loading skeletons as well

https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming

Reply:
Will use the `loading-screen` component which was implemented.